algos/demux/demux.rst

```diff
@@ -0,0 +1,138 @@
+.. _demux:
+
+Multiplexer/Demultiplexer
+########################
+
+Introduction
+************
+
+Multiplexer/demultiplexer component copies its input audio channels
+into output audio channels according to specific routing
+matrix. Multiplexer has multiple input audio streams and a single
+audio output stream. Demultiplexer has a single input stream and
+multiple output streams. In SOF codebase multiplexer and demultiplexer
+are implemented in a single component as the operations and
+configurations are overlapping heavily.
+
+.. figure:: images/muxdemux.png
+
+   Multiplexer has exactly 1 output stream and demultiplexer has exactly
+   1 input stream.
+
+Configuration
+=============
+
+Component configuration defines how audio channels are copied from
+input to output stream. As ASoC/SOF audio stream can have up to 8
+audio channels, a stream-to-stream specific 8 x 8 routing matrix
+defines the channel mapping from input to output. Because every stream
+is fully configurable we have a matrix for all multiplexer input
+streams or all demultiplexer output streams. 8 x 8 binary matrix takes up
+to 64 bits and is controlled with eight unsigned char values.
+
+.. figure:: images/mux.png
+
+   Example of multiplexer configuration matrices with 2 input streams.
+   In this artificial Mux example Input stream 1’s channel 1 is copied
+   to both output channels, Input stream 2’s single channel is copied
+   also to output channel 1.
```

Comment:
@juimonen I'm a newer to mux/demux. And I find it's a little hard for me to understand to understand the picture. Why do we need set column 3, 5, 7 to 1? Shouldn't it be 0s although it doesn't take efforts here. Or we must set 1s in column 3, 5, 7?

Reply:
@libinyang just think as the row as input channel, and 0/1 as if this row (channel) is put to column output channel. so for example first row (0,1,0,1,0,1,0,1) would mean that copy input channel 1 to output channels 2, 4, 6, 8. And so on for other channels. There's always 1 matrix per stream-to-stream mapping.